### PR TITLE
[WIP] restore `.py` file extensions to the tensorflow files

### PR DESCRIPTION
**Pull request by f0xxx1**
### **Describe your change:**

Whenever Tensorflow is released for python 3.9 (https://github.com/tensorflow/tensorflow/issues/44485) ensure that the tests are passing.

fixes: #3946 

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Python files are placed inside an existing directory.
* [ ] All filenames are in all lowercase characters with no spaces or dashes.
* [ ] All functions and variable names follow Python naming conventions.
* [ ] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by dhruvmanila** [comment]
Oh, there's one more PR to do the same job. I will keep this open. One more change  which needs to be  done is to remove the `python < 3.9` condition from `requirements.txt` for `tensorflow` and `keras`. Then, the tests should pass and this will be merged.

**Comment by dhruvmanila** [comment]
Ops, sorry for that.

**Comment by cclauss** [comment]
I believe that this will fail our tests because of #3946

**Comment by dhruvmanila** [comment]
I think there should only be one PR to fix #3946 and the rest should be closed in favor of it. This should be that PR as it already has the renaming part and we only need to test it after we enable `tensorflow`.

**Comment by dhruvmanila** [comment]
Oh, completely missed that support for Python 3.9 is only added for the release candidate version. I will test it out whether it works or not for that version.

**Comment by dhruvmanila** [comment]
Awesome, it's working now! Just needs to fix those pre-commit error and once the library is released, this should be good to go :)

@f0xxx1 Can you run `pre-commit` locally and fix those errors?

**Comment by f0xxx1** [comment]
@dhruvmanila I tried to fix the pre-commit errors however I don't think that black/flake8 are being compatible with the current configurations. 

I made some changes, but I think that I have to change way too much in order for it to work. Here is the output of the `pre-commit`

```
flake8...................................................................Failed
- hook id: flake8
- exit code: 1

scripts/validate_solutions.py:76:22: E701 multiple statements on one line (colon)
scripts/validate_solutions.py:76:22: E231 missing whitespace after ':'
scripts/validate_solutions.py:76:23: E225 missing whitespace around operator
dynamic_programming/k_means_clustering_tensorflow.py:39:9: E265 block comment should start with '# '
dynamic_programming/k_means_clustering_tensorflow.py:41:9: E265 block comment should start with '# '
dynamic_programming/k_means_clustering_tensorflow.py:42:9: E265 block comment should start with '# '
dynamic_programming/k_means_clustering_tensorflow.py:46:9: E265 block comment should start with '# '
dynamic_programming/k_means_clustering_tensorflow.py:47:9: E265 block comment should start with '# '
dynamic_programming/k_means_clustering_tensorflow.py:53:9: E265 block comment should start with '# '
dynamic_programming/k_means_clustering_tensorflow.py:54:9: E265 block comment should start with '# '
dynamic_programming/k_means_clustering_tensorflow.py:56:9: E265 block comment should start with '# '
dynamic_programming/k_means_clustering_tensorflow.py:57:9: E265 block comment should start with '# '
dynamic_programming/k_means_clustering_tensorflow.py:63:9: E265 block comment should start with '# '
dynamic_programming/k_means_clustering_tensorflow.py:70:9: E265 block comment should start with '# '
dynamic_programming/k_means_clustering_tensorflow.py:76:9: E265 block comment should start with '# '
dynamic_programming/k_means_clustering_tensorflow.py:77:9: E265 block comment should start with '# '
dynamic_programming/k_means_clustering_tensorflow.py:82:9: E265 block comment should start with '# '
dynamic_programming/k_means_clustering_tensorflow.py:84:9: E265 block comment should start with '# '
dynamic_programming/k_means_clustering_tensorflow.py:85:9: E265 block comment should start with '# '
dynamic_programming/k_means_clustering_tensorflow.py:86:9: E265 block comment should start with '# '
dynamic_programming/k_means_clustering_tensorflow.py:87:9: E265 block comment should start with '# '
dynamic_programming/k_means_clustering_tensorflow.py:93:9: E265 block comment should start with '# '
dynamic_programming/k_means_clustering_tensorflow.py:101:13: E265 block comment should start with '# '
dynamic_programming/k_means_clustering_tensorflow.py:102:13: E265 block comment should start with '# '
dynamic_programming/k_means_clustering_tensorflow.py:103:13: E265 block comment should start with '# '
dynamic_programming/k_means_clustering_tensorflow.py:109:17: E265 block comment should start with '# '
dynamic_programming/k_means_clustering_tensorflow.py:125:13: E265 block comment should start with '# '
```

Should I go for it or ?

**Comment by cclauss** [comment]
Please correct the flake8 issues.  Black did a new release in the past 48 hours so please ensure that .pre-commit is using the latest black release.

File: requirements.txt [hunk 1]
```diff
@@ -13,5 +13,5 @@ scikit-fuzzy
 sklearn
 statsmodels
 sympy
-tensorflow; python_version < '3.9'
+tensorflow>=2.5rc2
```
Comment (cclauss) [inline requirements.txt:16]: https://github.com/tensorflow/tensorflow/releases
```suggestion
tensorflow>=2.5rc3
```

File: neural_network/input_data.py [hunk 2]
```diff
@@ -307,30 +303,26 @@ def fake():
...
         test_labels_file, train_dir, source_url + test_labels_file
     )
     with gfile.Open(local_file, "rb") as f:
         test_labels = _extract_labels(f, one_hot=one_hot)
-
     if not 0 <= validation_size <= len(train_images):
         raise ValueError(
-            f"Validation size should be between 0 and {len(train_images)}. Received: {validation_size}."
+            f"Validation size should be between 0 "
+            f"and {len(train_images)}. Received: {validation_size}."
```
Comment (cclauss) [inline neural_network/input_data.py:324]:
```suggestion
            "Validation size should be between 0 and "
            f"{len(train_images)}. Received: {validation_size}."
```

File: neural_network/input_data.py [hunk 3]
```diff
@@ -46,16 +44,16 @@ def _read32(bytestream):
 def _extract_images(f):
```
Comment (cclauss) [inline neural_network/input_data.py:44]: Single letter variable names make the code look old school.  Would it be possible upgrade to more self-documenting variable names and add type hints?

**Comment by cclauss** [comment]
Please resolve conflicts because tensorflow v2.5 is shipping and #4422 has landed.

**Pull request closed without merging**